core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -427,5 +433,356 @@ namespace ledger {
+                    makeStringValue(unwrappedMessage.validatorDestinationAddress, allocator).Move());
+            const auto amount = JsonObject(cosmos::constants::kAmount,
+                    makeAmount(unwrappedMessage.amount, allocator).Move());
+
+            const auto rawTransaction = makeJsonFrom(document, baseReq, delegatorAddress,
+                    validatorSourceAddress, validatorDestinationAddress, amount);
+            const auto endpoint = fmt::format(cosmos::constants::kGaiaRedelegationsEndpoint,
+                    unwrappedMessage.delegatorAddress);
+
+            return genericPostRequestForSimulation(endpoint, rawTransaction);
+        }
+
+        Future<BigInt>
+        GaiaCosmosLikeBlockchainExplorer::getEstimatedGasLimit(
+                const std::shared_ptr<api::CosmosLikeTransaction>& transaction,
+                const std::shared_ptr<api::CosmosLikeMessage>& message) const
+        {
+            switch (message->getMessageType()) {
+                case api::CosmosLikeMsgType::MSGSEND:
+                    return getEstimatedGasLimitForTransfer(transaction, message);
+                case api::CosmosLikeMsgType::MSGWITHDRAWDELEGATORREWARD:
+                    return getEstimatedGasLimitForRewards(transaction, message);
+                case api::CosmosLikeMsgType::MSGDELEGATE:
+                    return getEstimatedGasLimitForDelegations(transaction, message);
+                case api::CosmosLikeMsgType::MSGUNDELEGATE:
+                    return getEstimatedGasLimitForUnbounding(transaction, message);
+                case api::CosmosLikeMsgType::MSGBEGINREDELEGATE:
+                    return getEstimatedGasLimitForRedelegations(transaction, message);
+                default:
+                    return Future<BigInt>(nullptr)
+                        .map<BigInt>(getContext(), [](BigInt) {
+                                return BigInt::ZERO;
+                        });
+            }
+        }
+
+        FuturePtr<BigInt>
+        GaiaCosmosLikeBlockchainExplorer::getEstimatedGasLimit(
+                const std::shared_ptr<api::CosmosLikeTransaction>& transaction) const
+        {
+            const auto& messages = transaction->getMessages();
+            auto estimations = std::vector<Future<BigInt>>();
+            std::transform(std::begin(messages), std::end(messages),
+                    std::back_inserter(estimations),
+                    [&](const std::shared_ptr<api::CosmosLikeMessage>& message) {
+                        return getEstimatedGasLimit(transaction, message);
+                    });
+
+            return async::sequence(getContext(), estimations)
+                .flatMapPtr<BigInt>(getContext(), [](const auto& estimations) {
+                        const auto result = std::accumulate(std::begin(estimations),
+                                std::end(estimations), BigInt::ZERO);
```

Comment:
Can you just add a comment to say that signature verification cost is accumulated 1 time per message in the transaction, but that we don't care because it's usually a lot smaller than the rest of the gas cost ?

I'm afraid we will forget otherwise (or maybe I'm wrong and I forgot what we decided about fixed gas costs)

Reply:
I can do that, or I can remove the cost of the signature (with the constant value in the cosmos repo) for every extra message. Just tell me what you prefer.

Reply:
Let's ignore the cost for now, common transactions use amounts in the `x e4` Gas, the signature cost can be ignored I think